.github/workflows/vercel.yml

```diff
@@ -7,13 +7,18 @@ env:
 on:
   push:
     branches:
+      - master
       - beta
   pull_request:
 
 jobs:
   vercel:
     runs-on: ubuntu-latest
 
+    environment:
+      name: ${{ github.ref_name == 'master' && 'Production' || 'Preview' }}
```

Comment:
Not sure this will work according to the [docs](https://docs.github.com/en/actions/using-jobs/using-environments-for-jobs):
> The URL can be an expression and can use any context except for the [secrets context](https://docs.github.com/en/actions/learn-github-actions/contexts#contexts).

We can merge and try it if you'd like. If it doesn't work, you may need to split this out into two separate jobs, `vercel-preview` and `vercel-prod`, and use [job conditions](https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution) to determine when each job should run based on the `github.ref_name` value.

Reply:
Ok sounds good.

The current preview build actually worked and tagged the deployment correctly: https://github.com/hirosystems/token-metadata-service/deployments/activity_log?environment=Preview
but it remains to see if the `Production` conditional will work as expected. If it doesn't, I'll split it into two jobs (I wanted to avoid DRY as much as possible)

Reply:
Nice, then I'd say let's merge and try it!